climada/hazard/centroids/centr.py

```diff
@@ -555,60 +555,48 @@ def union(self, *others):
         ------
             ValueError
         """
-        cent_list = list(others)
-        if self.lat.any() or self.meta:
-            cent_list.insert(0, self)
-
-        if np.all([cent_list[0].equal(cent) for cent in cent_list[1:]]):
-            return copy.deepcopy(cent_list[-1])
-
-        #Set attributes to centroids if missing in one but defined in other
-        is_area_pixel = np.any([cent.area_pixel.any() for cent in cent_list]) \
-            and not np.all([cent.area_pixel.any() for cent in cent_list])
-        is_dist_coast = np.any([cent.dist_coast.any() for cent in cent_list]) \
-            and not np.all([cent.dist_coast.any() for cent in cent_list])
-        is_on_land = np.any([len(cent.on_land) > 0 for cent in cent_list]) \
-            and not np.all([len(cent.on_land) > 0 for cent in cent_list])
-        is_region_id = np.any([cent.region_id.any() for cent in cent_list]) \
-            and not np.all([cent.region_id.any() for cent in cent_list])
-        is_elevation = np.any([cent.elevation.any() for cent in cent_list]) \
-            and not np.all([cent.elevation.any() for cent in cent_list])
+        # restrict to non-empty centroids
```

Comment:
Since this function's operation is like the union of sets, empty centroid objects should not change the result at all. We remove them in the very first step to avoid manual checks for emptiness of objects later in the code.